velox/common/tests/BloomTest.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/Bloom.h"
+#include <folly/Random.h>
+#include <unordered_set>
+
+#include <gtest/gtest.h>
+
+using namespace facebook::velox;
+TEST(BloomTest, basic) {
+  constexpr int32_t kWords = 128;
+  constexpr int32_t kSize = kWords * 8;
+  folly::Random::DefaultGenerator rng;
+  rng.seed(1);
+  std::unordered_set<uint64_t> reference;
+  std::vector<uint64_t> bits(128);
+  // We insert kSize random values and check that they are there.
+  for (auto i = 0; i < kSize; ++i) {
+    auto value = folly::Random::rand64(rng);
+    Bloom::set(bits.data(), kWords, value);
+    reference.insert(value);
+  }
+  for (auto value : reference) {
+    EXPECT_TRUE(Bloom::test(bits.data(), kWords, value));
+  }
+  int32_t hits = 0;
+  for (auto i = 0; i < kSize; ++i) {
+    auto value = folly::Random::rand64(rng);
+    hits += Bloom::test(bits.data(), kWords, value);
+  }
+  EXPECT_GT(45, hits);
+}
+
+TEST(BloomTest, precision) {
+  constexpr int32_t kSize = 1024;
+  Bloom bloom(kSize);
+  for (auto i = 0; i < kSize; ++i) {
+    bloom.insert(i);
+  }
+  int32_t numFalsePositives = 0;
+  for (auto i = 0; i < kSize; ++i) {
+    EXPECT_TRUE(bloom.mayContain(i));
+    numFalsePositives += bloom.mayContain(i + kSize);
+    numFalsePositives += bloom.mayContain((i + kSize) * 123451);
+  }
+  EXPECT_GT(kSize / 50, numFalsePositives / 2);
```

Comment:
I find this test very difficult to understand. I assume this is assert that number of false positives is < 2% of total checks, right? Perhaps, it would be cleaner to express this as EXPECT_LT(numFalsePositives, (2 * kSize) * 0.02)